pkg/image/admission/imagepolicy/imagepolicy_test.go

```diff
@@ -429,12 +480,20 @@ func TestAdmissionResolution(t *testing.T) {
 		pod.Spec.Containers[1].Image != "myregistry.com/mysql/mysql@sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4" {
 		t.Errorf("unexpected image: %#v", pod)
 	}
+	if err := p.Validate(attrs); err != nil {
+		t.Logf("object: %#v", attrs.GetObject())
+		t.Fatal(err)
+	}
+	if pod.Spec.Containers[0].Image != "myregistry.com/mysql/mysql@sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4" ||
+		pod.Spec.Containers[1].Image != "myregistry.com/mysql/mysql@sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4" {
+		t.Errorf("unexpected image: %#v", pod)
+	}
 }
 
 func TestAdmissionResolveImages(t *testing.T) {
 	image1 := &imageapi.Image{
 		ObjectMeta:           metav1.ObjectMeta{Name: "sha256:0000000000000000000000000000000000000000000000000000000000000001"},
-		DockerImageReference: "integrated.registry/image1/image1:latest",
+		DockerImageReference: "integrated.registry/image1/image1@sha256:0000000000000000000000000000000000000000000000000000000000000001",
```

Comment:
this was bad test data... in real life, all `Image` objects have a `DockerImageReference` containing a sha, not a tag... putting a tag here meant the build referencing a specific sha got resolved to a tag (which is wrong), and then tried to re-resolve to a sha during the validation stage